File: jlox/lox/TokenList.java
package lox;

import java.util.ArrayList;
import java.util.List;

import java.util.Iterator;

public class TokenList {

	private final List<Token> tokens;

	TokenList() {
		this.tokens = new ArrayList<>();
	}

	public int size() {
		return this.tokens.size();
	}

	public String asString() {
		StringBuilder sb = new StringBuilder();
		Iterator<Token> iter = this.tokens.iterator();
		sb.append('[');
		while (iter.hasNext()) {
			sb.append(iter.next().asString());
			if (iter.hasNext()) {
				sb.append(", ");
			}
		}
		sb.append(']');
		return sb.toString();
	}

	public void addToken(Token t) {
		this.tokens.add(t);
	}

}

File: CSharpLox/TokenList.cs
using System;
using System.Collections.Generic;

namespace CSharpLox
{
	public class TokenList
	{
		public int Count { get => this.tokens.Count; }

		private readonly List<Token> tokens;

		public TokenList()
		{
			this.tokens = new List<Token>();
		}

		public override string ToString()
		{
			return $"[{String.Join(", ", this.tokens)}]";
		}

		public void Add(Token t)
		{
			this.tokens.Add(t);
		}

	}
}